### PR TITLE
Allow running deterministic tests

### PR DESCRIPTION
**Pull request by hobarrera**
Make tests deterministic when the `DETERMINISTIC_TESTS` environment variable is set to true. This satisfies requirements of downstream packagers which require deterministic tests.

Also run larger test samples when running on CI, in hope that at some point it might trigger a new edge case.

See #352
Would probably require an update to #354

### CODE REVIEW

**Comment by untitaker** [comment]
Could you add a default variable to the `Makefile`, and don't fallback to a default in Python code (just to be consistent with other envvars)?

File: tests/conftest.py [hunk 1]
```diff
@@ -27,3 +28,16 @@ def benchmark():
...
+    max_examples=1000,
+    verbosity=Verbosity.verbose,
+))
+settings.register_profile("deterministic", settings(
+    derandomize=True,
+))
+
+if os.getenv('DETERMINISTIC_TESTS', False):
+    settings.load_profile("deterministic")
+elif os.getenv('CI', False):
```
Comment (untitaker) [inline tests/conftest.py]: Please compare both variables to `"true"`.

**Comment by hobarrera** [comment]
Fixed both issues.

**Comment by untitaker** [comment]
Merged into #354, thanks @hobarrera!
